packages/core/lib/console.js

```diff
@@ -144,15 +132,49 @@ class Console extends EventEmitter {
       contextVars[abstraction.contract_name] = abstraction;
     });
 
-    this.repl.setContextVars(contextVars);
+    // make sure the repl gets the new contracts in its context
+    Object.keys(contextVars || {}).forEach(key => {
+      this.repl.context[key] = contextVars[key];
+    });
+  }
+
+  runSpawn(inputStrings, options, callback) {
```

Comment:
We should rewrite this method to not take a callback. This does not hold up this PR from being merged, I just wanted to note that here.